src/Database/PostgreSQL/TransactionalStore.hs

```diff
@@ -21,74 +22,97 @@ module Database.PostgreSQL.TransactionalStore
 import           Control.Applicative
 #endif
 import           Control.Monad.Reader
+import           Control.Monad.Trans.Control
 import           Data.Int
-import qualified Database.PostgreSQL.Simple         as Postgres
-import qualified Database.PostgreSQL.Simple.Transaction as Postgres.Transaction
-import qualified Database.PostgreSQL.Simple.Types   as PGTypes
+import qualified Database.PostgreSQL.Simple             as Postgres
 import           Database.PostgreSQL.Simple.FromRow
 import           Database.PostgreSQL.Simple.ToRow
+import qualified Database.PostgreSQL.Simple.Transaction as Postgres.Transaction
+import qualified Database.PostgreSQL.Simple.Types       as PGTypes
 
-newtype PGTransaction a =
-    PGTransaction (ReaderT Postgres.Connection IO a)
+newtype PGTransactionT m a =
+    PGTransactionT (ReaderT Postgres.Connection m a)
     deriving ( Functor
              , Applicative
              , Monad
-             , MonadIO
+             , MonadTrans
              , MonadReader Postgres.Connection
+             , MonadIO
              )
 
-runPGTransaction' :: MonadIO m
+type PGTransaction = PGTransactionT IO
+
+runPGTransaction' :: MonadBaseControl IO m
```

Comment:
Chewing on this a bit more, I wonder if I may have misunderstood your question. Why is the return type not `MonadIO m => m a`? The function  `control` works to extract an `IO a` value from the `ReaderT`, passes this value to `withTransactionLevel`, and finally 'wraps' up thist the returned `IO a` into  `ReaderT`. So, there's no values in `IO` that we need to `liftIO`!

Thinking on this did make me realize that this _can be_ a breaking change for some users, if they were expecting the return type to be something which implements `MonadIO` but not `MonadBaseControl`. But this is a fairly easy fix, as `IO` implements `MonadBaseControl`, so you can just `liftIO` the result of `runPGTransaction` when `m ~ IO`